### PR TITLE
Added readme and contribution guidelines for the repo.

### PR DESCRIPTION
**Pull request by aaithal**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: go
+sudo: false
+go:
+  - 1.7
```
Comment (samuelkarp) [inline .travis.yml]: This should be `1.7.x`.  `1.7` just uses 1.7.0: https://docs.travis-ci.com/user/languages/go/#Specifying-a-Go-version-to-use

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+Any contributions should pass all tests, including those not run by our
+current CI system.
+
+You may run all test by either running the `make test` target (requires `go`,
+and `go cover` to be installed).
+
+## Licensing
+
+The Amazon ECS Agent is released under an [Apache
```
Comment (samuelkarp) [inline CONTRIBUTING.md]: This isn't the agent repo. :smile:

File: README.md [hunk 3]
```diff
@@ -1 +1,15 @@
-TODO
+# Amazon ECS CNI Plugins
+
+[![Build Status](https://travis-ci.org/aws/amazon-ecs-cni-plugins.svg?branch=dev)](https://travis-ci.org/aws/amazon-ecs-cni-plugins)
```
Comment (samuelkarp) [inline README.md]: `dev` or `master`?  Maybe leave it as `dev` now and change to `master` right before release?

Comment (aaithal) [inline README.md]: `dev` for now until we merge code into `master`, which I think we should do once this PR is merged.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Development takes place against the `dev` branch of this repository and pull
+requests should be opened against that branch.
+
+### Testing
+
+Any contributions should pass all tests, including those not run by our
+current CI system.
+
+You may run all test by either running the `make test` target (requires `go`,
```
Comment (richardpen) [inline CONTRIBUTING.md]: either ... or .. ?

**Comment by aaithal** [comment]
@samuelkarp I've already asked @nrdlngr to have a look at this.

File: README.md [hunk 5]
```diff
@@ -1 +1,15 @@
...
+
+Amazon ECS CNI Plugins is a collection of Container Network Interface([CNI](https://github.com/containernetworking/cni)) Plugins used by the [Amazon ECS Agent](https://github.com/aws/amazon-ecs-agent) to configure network namespace of containers with Elastic Network Interfaces ([ENIs](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html))
+
+For more information about Amazon ECS, see the [Amazon ECS Developer Guide](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/Welcome.html).
+
+For more information about Plugins in this project, see the individual READMEs.
+
+## Plugins
+* [ECS ENI Plugin](plugins/eni/README.md): configures the network namespace of the container with an ENI device
+* [ECS Bridge Plugin](plugins/ecs-bridge/README.md): configures the network namespace of the container to be able to communicate with the credentials endpoint of the ECS Agent
```
Comment (richardpen) [inline README.md]: how about: `set up the bridge and veth pair in the container network namespace to be able to ...` or better to include bridge setting up here.

Comment (aaithal) [inline README.md]: I don't want to go into too much detail here as the README for bridge is linked anyway. So, would not want to add more things here, unless you feel strongly that we should.

File: plugins/ipam/README.md [hunk 6]
```diff
@@ -1,8 +1,8 @@
-# IPAM plugin
+# ECS IPAM plugin
 
 ## Overview
 
-The IPAM plugin will construct the IP, Gateway, Routes as a struct which will be
+The ECS IPAM plugin will construct the IP, Gateway, Routes as a struct which will be
```
Comment (nrdlngr) [inline plugins/ipam/README.md]: The ECS IPAM plugin constructs the IP, Gateway, and Routes, which are used as a struct

File: plugins/ipam/README.md [hunk 7]
```diff
@@ -1,8 +1,8 @@
...
 used by the ECS Bridge plugin to configure the bridge and veth pair in the
```
Comment (nrdlngr) [inline plugins/ipam/README.md]: by the ECS Bridge plugin to configure the bridge and veth pair in the

**Comment by aaithal** [comment]
@nrdlngr I have modified it as per your comments. Can you please take another look?

**Pull request merged**